jinxify/src/app/(protected-app)/dashboard/page.tsx

```diff
@@ -4,206 +4,209 @@ import { useQuery, useMutation, useQueryClient } from "@tanstack/react-query";
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          type,
+          title,
+          directoryId: currentDirectory?.id,
+        }),
+      });
 
-			if (!response.ok) {
-				throw new Error("Failed to create workspace item");
-			}
+      if (!response.ok) {
+        throw new Error("Failed to create workspace item");
+      }
 
-			return response.json();
-		},
-		onSuccess: () => {
-			// Invalidate and refetch the current directory data
-			queryClient.invalidateQueries({ queryKey: ["directory", directoryId] });
-		},
-	});
+      return response.json();
+    },
+    onSuccess: () => {
+      // Invalidate and refetch the current directory data
+      queryClient.invalidateQueries({ queryKey: ["directory", directoryId] });
+    },
+  });
 
-	const handleCreateFolder = () => {
-		createWorkspaceItem.mutate({
-			type: "directory",
-			title: "New Folder", // You might want to add a dialog to get the title from user
-		});
-	};
+  const handleCreateFolder = () => {
+    createWorkspaceItem.mutate({
+      type: "directory",
+      title: "New Folder", // You might want to add a dialog to get the title from user
+    });
+  };
 
-	const handleCreateDiagram = () => {
-		createWorkspaceItem.mutate({
-			type: "diagram",
-			title: "New Diagram", // You might want to add a dialog to get the title from user
-		});
-	};
+  const handleCreateDiagram = () => {
+    createWorkspaceItem.mutate({
+      type: "diagram",
+      title: "New Diagram", // You might want to add a dialog to get the title from user
+    });
+  };
 
-	if (isLoading) {
-		return <div>Loading...</div>;
-	}
+  if (isLoading) {
+    return <div>Loading...</div>;
+  }
 
-	const items = [
-		...(currentDirectory?.directories ?? []),
-		...(currentDirectory?.diagrams ?? []),
-	];
+  const items = [
+    ...(currentDirectory?.directories ?? []),
+    ...(currentDirectory?.diagrams ?? []),
+  ];
 
-	return (
-		<div className="flex h-screen w-screen bg-gray-50">
-			<AppSidebar />
-			<main className="flex-1 flex flex-col ml-4 mr-4 mt-4 mb-4 p-4 bg-white shadow-xl rounded-lg border border-gray-200">
-				<div className="flex justify-between items-center mb-6">
-					<h1 className="text-2xl font-bold">
-						{currentDirectory?.title || "My Workspace"}
-					</h1>
-					<SidebarTrigger className="md:hidden" />
-					<div className="space-x-2">
-						<Button
-							disabled={createWorkspaceItem.isPending}
-							variant="outline"
-							onClick={handleCreateFolder}
-						>
-							<FaFolder className="w-5 h-5 text-purple-1000" />
-						</Button>
-						<Button
-							disabled={createWorkspaceItem.isPending}
-							onClick={handleCreateDiagram}
-						>
-							<FaRegFileAlt className="w-5 h-5" />
-						</Button>
-					</div>
-				</div>
+  return (
+    <div className="flex h-screen w-screen bg-gray-50">
+      <AppSidebar session={session ?? null} />
```

Comment:
Why dont we just get the session directly on the AppSidebar instead of passing it as props

Reply:
Tried that but there was an Error which  didn't the content of page to render